frontends/PyCDE/src/pycde/module.py

```diff
@@ -204,16 +208,21 @@ def __call__(self, op):
     }
 
     # Build the replacement HWModuleOp in the outer module.
+    module_key = str((op.name, sorted(op_types), sorted(result_types),
+                      sorted(params.items())))
     with mlir.ir.InsertionPoint(mod.regions[0].blocks[0]):
```

Comment:
If you change the order of this `with` and the contained `if`, we could avoid creating the `InsertionPoint`.